common/components/BalanceSidebar/BalanceSidebar.jsx

```diff
@@ -0,0 +1,151 @@
+import type { TokenBalance } from 'selectors/wallet';
+import { getNetworkConfig } from 'selectors/config';
+import { Link } from 'react-router';
+import TokenBalances from './TokenBalances';
+import { formatNumber } from 'utils/formatters';
+import { Identicon } from 'components/ui';
+import translate from 'translations';
+import * as customTokenActions from 'actions/customTokens';
+
+type Props = {
+  wallet: BaseWallet,
+  balance: Big,
+  network: NetworkConfig,
+  tokenBalances: TokenBalance[],
+  addCustomToken: typeof customTokenActions.addCustomToken,
+  removeCustomToken: typeof customTokenActions.removeCustomToken
+};
+
+export class BalanceSidebar extends React.Component {
+  props: Props;
+  state = {
+    showLongBalance: false
+  };
+
+  render() {
+    const { wallet, balance, network, tokenBalances } = this.props;
+    const { blockExplorer, tokenExplorer } = network;
+    if (!wallet) {
+      return null;
+    }
+    const ajaxReq = {};
+
+    return (
+      <aside>
+        <h5>
+          {translate('sidebar_AccountAddr')}
+        </h5>
+        <ul className="account-info">
+          <Identicon address={wallet.getAddress()} />
+          <span className="mono wrap">
+            {wallet.getAddress()}
+          </span>
+        </ul>
+        <hr />
+        <h5>
+          {translate('sidebar_AccountBal')}
+        </h5>
+        <ul
+          className="account-info point"
+          onDoubleClick={this.toggleShowLongBalance}
+          title={`${balance.toString()} (Double-Click)`}
+        >
+          <li>
+            <span className="mono wrap">
+              {this.state.showLongBalance ? balance.toString() : formatNumber(balance)}
+            </span>
+            {` ${network.name}`}
+          </li>
+        </ul>
+        <TokenBalances
+          tokens={tokenBalances}
+          onAddCustomToken={this.props.addCustomToken}
+          onRemoveCustomToken={this.props.removeCustomToken}
+        />
+        <hr />
+        {(!!blockExplorer || !!tokenExplorer) &&
+          <div>
+            <h5>
+              {translate('sidebar_TransHistory')}
+            </h5>
+            <ul className="account-info">
+              {!!blockExplorer &&
+                <li>
+                  <a
+                    href={blockExplorer.address.replace('[[address]]', wallet.getAddress())}
+                    target="_blank"
+                  >
+                    {`${network.name} (${blockExplorer.name})`}
+                  </a>
+                </li>}
+              {!!tokenExplorer &&
+                <li>
+                  <a
+                    href={tokenExplorer.address.replace('[[address]]', wallet.getAddress())}
+                    target="_blank"
+                  >
+                    {`Tokens (${tokenExplorer.name})`}
+                  </a>
+                </li>}
+            </ul>
+          </div>}
+        <hr />
+        {false &&
```

Comment:
You should leave a `// TODO` comment with an explanation for what this section does.